include/exadg/incompressible_navier_stokes/spatial_discretization/spatial_operator_base.cpp

```diff
@@ -360,11 +360,22 @@ SpatialOperatorBase<dim, Number>::fill_matrix_free_data(
                                      field + quad_index_u_overintegration);
 
   // TODO create these quadrature rules only when needed
-  matrix_free_data.insert_quadrature(dealii::QGaussLobatto<1>(param.degree_u + 1),
-                                     field + quad_index_u_gauss_lobatto);
-  matrix_free_data.insert_quadrature(dealii::QGaussLobatto<1>(param.get_degree_p(param.degree_u) +
-                                                              1),
-                                     field + quad_index_p_gauss_lobatto);
+  if(param.grid.element_type == ElementType::Hypercube)
+  {
+    matrix_free_data.insert_quadrature(dealii::QGaussLobatto<1>(param.degree_u + 1),
+                                       field + quad_index_u_gauss_lobatto);
+    matrix_free_data.insert_quadrature(dealii::QGaussLobatto<1>(param.get_degree_p(param.degree_u) +
+                                                                1),
+                                       field + quad_index_p_gauss_lobatto);
+  }
+  else if(param.grid.element_type == ElementType::Simplex)
+  {
+    matrix_free_data.insert_quadrature(dealii::QSupportPointsSimplex<dim>(param.degree_u),
+                                       field + quad_index_u_gauss_lobatto);
```

Comment:
As suggested in the deal.II PR, can't you use `dof_handler_u.get_fe().get_unit_support_points()`?

Reply:
Unfortunately not, `MappingInfo` is trying to create a face quadrature, which fails for `Qudarature<dim>(dof_handler_u.get_fe().get_unit_support_points())` as the quadrature doesn't know any rule how to create the face quadrature.